File: PythonProjects/DataStructures/lib/singlearray.py
"""
This is an ADT which is used to create an Array
"""


class SingleArray:
    data = []
    used_size = 0

    def __init__(self, total_size):
        self.total_size = total_size
        self.data = [0]*total_size

    def print_array(self):
        print(self.data)

    def set_value(self,value):
        self.data[self.used_size] = value
        self.used_size = self.used_size + 1

    def get_total_size(self):
        return self.total_size

    def get_used_size(self):
        return self.used_size

File: Java Programs/lib/SingleArray.java
package lib;
public class SingleArray {

    public SingleArray(int tSize){
      this.totalSize = tSize;
      this.usedSize = 0;
      this.data = new int[tSize];
    }

    int totalSize,  usedSize;
    int [] data;

    public void setData(int data){
      this.data[this.usedSize]=data;
      this.usedSize++;
    }

    public void showArray(){
        for(int i=0;i<usedSize;i++){
            System.out.println(this.data[i]);
        }
    }
    public int getTotalSize(){
        return this.totalSize;
    }

    public int getUsedSize(){
        return this.usedSize;
    }

    public int[] getData(){
        return this.data;
    }
}
